BootloaderCommonPkg/Library/ContainerLib/ContainerLib.c

```diff
@@ -395,11 +399,12 @@ LocateComponentRegion (
 **/
 EFI_STATUS
 EFIAPI
-LoadComponent (
-  IN     UINT32    ContainerSig,
-  IN     UINT32    ComponentName,
-  IN OUT VOID    **Buffer,
-  IN OUT UINT32   *Length
+LoadComponentWithCallback (
+  IN     UINT32                   ContainerSig,
+  IN     UINT32                   ComponentName,
+  IN OUT VOID                   **Buffer,
+  IN OUT UINT32                  *Length,
+  IN     LOAD_COMPONENT_CALLBACK  LoadComponentCallback
   )
```

Comment:
I had thought about it.   But I still think this way is better.  This library should have nothing to do with performance.  And it should not have to understand the performance data format.  It is cleaner.

Reply:
It is not necessary to understand performance data format since only measurement ID is required. The PerfIdBase benefit is that we don't need add callback function if you want to measure other components (e.g. fastboot, IPFW, etc) performance. Anyway, it is fine if you think callback is better.
Another one is that: since last parameter is optional, do we still need expose another API LoadComponent(), or just rename this API to LoadComponent()?

Reply:
If you want this library to fill in the performance data,  then it needs to understand where to fill it, how to get performance data.  To me, these are not necessary related to this lib.  I would only imagine the LoadComponentWithCallback() to be called in Stage1B and Stage2 to log the performance data.   Other places normally will just call LoadComponent()  instead.    These is why I still keep the wrapper function  available to make it simpler for other cases.